zipp.py

```diff
@@ -62,6 +62,46 @@ def _difference(minuend, subtrahend):
     return itertools.filterfalse(set(subtrahend).__contains__, minuend)
 
 
+class PickleableClass:
+    """
+    Utility object that wraps another un-pickleable object that,
+    for example, might hold a file object, and saves the
+    initialization parameters. When pickeled, the un-pickleable
+    object is discarded, and when loaded, it is rebuilt from the
+    initialization params."""
+
+    _class: type = object
+
+    def __init__(self, *args, **kwargs):
+        self._args = args
+        self._kwargs = kwargs
+        self._obj = None
+
+    def __getattr__(self, name):
+        if self._obj is None:
+            self._obj = self._class(*self._args, **self._kwargs)
+        return getattr(self._obj, name)
+
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state['_obj'] = None
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__.update(state)
+
+    def __repr__(self):
+        args = f"class={self._class}, args={self._args}, kwargs={self._kwargs}"
+        return f"{self.__class__.__name__}({args})"
+
+    def __eq__(self, other):
+        return (
+            self._class == other._class
+            and self._args == other._args
+            and self._kwargs == other._kwargs
+        )
```

Comment:
I don't care for the way that this wrapper changes the repr and eq of the object it wraps.

Reply:
In 6f700a439e7535878fdc5639e6368ee3a9691221, I've limited the pickling support to a mix-in that largely doesn't affect other behaviors.